Framework/PythonInterface/plugins/algorithms/BASISReduction.py

```diff
@@ -454,41 +456,66 @@ def _sum_runs(self, run_set, sam_ws, extra_ext=None):
             ws_name = self._make_run_name(run)
             if extra_ext is not None:
                 ws_name += extra_ext
-            run_file = self._make_run_file(run)
-
-            sapi.LoadEventNexus(Filename=run_file,
-                                OutputWorkspace=ws_name,
-                                BankName=self._reflection['banks'])
-            if str(run)+':' in self.getProperty('ExcludeTimeSegment').value:
-                self._filterEvents(str(run), ws_name)
+            self.load_single_run(run, ws_name)
             if sam_ws != ws_name:
                 sapi.Plus(LHSWorkspace=sam_ws,
                           RHSWorkspace=ws_name,
                           OutputWorkspace=sam_ws)
                 sapi.DeleteWorkspace(ws_name)
 
+    def load_single_run(self, run, name):
+        """
+        Find and load events.
+
+        Applies event filtering if necessary.
+
+        Parameters
+        ----------
+        run: str
+            Run number
+        name: str
+            Name of the output EventsWorkspace
+
+        Returns
+        -------
+        EventsWorkspace
+        """
+
+        kwargs = dict(Filename=self._make_run_file(run),
+                      BankName=self._reflection['banks'],
+                      OutputWorkspace=name)
+        if str(run) + ':' in self.getProperty('RetainTimeSegment').value:
+            kwargs.update(self._retainEvents(run))
+        sapi.LoadEventNexus(**kwargs)
+        if str(run) + ':' in self.getProperty('ExcludeTimeSegment').value:
+            self._filterEvents(run, name)
+
     def _sum_flux(self, run_set, mon_ws, extra_ext=None):
-        self._aggregate_flux = 0.0
+        r"""
```

Comment:
Is there any reason this doc string is a string literal and the one above isn't?
As far as I'm aware it's only necessary if the string contains a backslash as Python3 often interprets this as an invalid escape sequence

Reply:
@TTitcombe No other than "historical" reasons. I have changed all docstrings to Numpy style, and using string literals